include/eve/arch/riscv/as_register.hpp

```diff
@@ -0,0 +1,299 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/riscv/predef.hpp>
+#include <eve/arch/riscv/rvv_utils.hpp>
+#include <eve/traits/as_integer.hpp>
+
+#include <type_traits>
+#if defined(EVE_INCLUDE_RISCV_HEADER)
+
+namespace eve
+{
+template<typename T> struct logical;
+}
+
+#  if defined(EVE_RETURN_M)
+#    error EVE_RETURN_M already defined
```

Comment:
In which scenario can this happens ?

Reply:
This is just to be safe. I've already had an experience of debugging override of macros, so I prefer to have a guarantee that this situation is not happening here. If you wish, I can delete this, as now we do not have any possibility to have this macro defined.